apps/modernization-ui/src/apps/page-builder/page/management/edit/question/GroupQuestion/GroupQuestion.tsx

```diff
@@ -0,0 +1,150 @@
+import styles from './group-question.module.scss';
+import { useForm, FormProvider } from 'react-hook-form';
+import { Form, ModalRef } from '@trussworks/react-uswds';
+import { RefObject, useState } from 'react';
+import { ModalToggleButton } from '@trussworks/react-uswds';
+import { RepeatingBlock } from './RepeatingBlock';
+import { SubsectionDetails } from './SubsectionDetails';
+import { SubSectionControllerService } from 'apps/page-builder/generated';
+import { authorization } from 'authorization';
+import { usePageManagement } from '../../../usePageManagement';
+
+type Props = {
+    subsection: PagesSubSection;
+    questions: PagesQuestion[];
+    modalRef: RefObject<ModalRef>;
+};
+
+type Additional = {
+    repeatNumber: number;
+    visibleText: string;
+};
+
+type GroupQuestionFormType = GroupSubSectionRequest & PagesSubSection & Additional;
+
+export const GroupQuestion = ({ subsection, questions, modalRef }: Props) => {
+    const { page, setAlert, refresh } = usePageManagement();
+    const [valid, setValid] = useState(false);
+    const init = {
+        name: subsection.name,
+        batches: questions.map((question) => ({
+            batchTableAppearIndCd: undefined,
+            batchTableColumnWidth: undefined,
+            batchTableHeader: undefined,
+            id: question.id
+        })),
+        blockName: undefined,
+        id: subsection.id,
+        visibleText: subsection.visible ? 'Y' : 'N',
+        repeatNumber: 1
+    };
+    const methods = useForm<GroupQuestionFormType>({
+        mode: 'onChange',
+        defaultValues: { ...init }
+    });
+    const { handleSubmit, reset } = methods;
+
+    const handleGroup = (data: GroupQuestionFormType) => {
+        const request = {
+            batches: data.batches,
+            blockName: data.blockName,
+            id: data.id
+        };
+        try {
+            SubSectionControllerService.groupSubSectionUsingPost({
+                authorization: authorization(),
+                page: page.id,
+                request: request
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>{error.message}</p>
+                });
+            } else {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>An unknown error occurred</p>
+                });
+            }
+        } finally {
+            setAlert({
+                type: 'success',
+                expiration: 5000,
+                message: (
+                    <p>
+                        You've successfully grouped <span>{data.blockName}</span>
+                    </p>
+                )
+            });
+            refresh();
+        }
```

Comment:
Issue: Doesn't `finally` always execute even if there is an error? If so that means that in the case of an error, an error message will appear followed by a success message.